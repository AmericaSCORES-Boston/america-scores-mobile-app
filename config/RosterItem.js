import React, { Component } from 'react';

import {
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';

const RosterItem = (props) => {
  const { name, id, people } = props.roster;
  const itemStyle = props.isSelected && [styles.item, styles.selected] || styles.item;
  return (
    <TouchableHighlight
      onPress={() => {props.selectRoster(id)}}
      underlayColor='#E0F2F1'
    >
      <View style={itemStyle}>
        <Text style={styles.title}>{name}</Text>
        <Text style={styles.stars}>{`${people} stars`}</Text>
      </View>
    </TouchableHighlight>
  );
};

RosterItem.propTypes = {
  isSelected: React.PropTypes.bool,
  roster: React.PropTypes.object.isRequired,
  selectRoster: React.PropTypes.func,
};

RosterItem.defaultProps = {
  isSelected: false,
};

export const styles = StyleSheet.create({
  item: {
    paddingVertical: 8,
    paddingHorizontal: 24,
  },
  selected: {
    backgroundColor: '#B2DFDB',
  },
  stars: {
    paddingBottom: 8,
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    paddingBottom: 4,
  },
});

export default RosterItem;
