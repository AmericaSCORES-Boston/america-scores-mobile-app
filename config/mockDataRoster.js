export const rosters = [
  {
    name: 'Test 1',
    id: 4,
    people: ['John Doe', 'Jane Doe'],
  },
  {
    name: 'Test 2',
    id: 5,
    people: ['Billy Bob'],
  },
  {
    name: 'Test 3',
    id: 6,
    people: ['John Michael', 'Harry Reynolds'],
  },
];
