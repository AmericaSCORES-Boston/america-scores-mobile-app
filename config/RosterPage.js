import React, { Component } from 'react';
import {
  ListView,
  StyleSheet,
  Text,
} from 'react-native';

import RosterItem from './RosterItem.js';

const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id});

const RosterPage = (props) => {
  const { rosters, selected, selectRoster } = props;
  const dataSource = ds.cloneWithRows(rosters);
  return (
    <ListView
      dataSource={dataSource}
      renderRow={(roster) => (
        <RosterItem
          isSelected={roster.id === selected}
          roster={roster}
          selectRoster={selectRoster}
        />
      )}
      contentContainerStyle={styles.list}
    />
  );
};

RosterPage.propTypes = {
  rosters: React.PropTypes.array.isRequired,
  selected: React.PropTypes.number,
  selectRoster: React.PropTypes.func,
};

export const styles = StyleSheet.create({
  list: {
    paddingVertical: 40, // (48 - 8 of item padding)
  }
});

export default RosterPage;
