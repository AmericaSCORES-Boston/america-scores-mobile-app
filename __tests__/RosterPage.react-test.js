import React from 'react';
import renderer from 'react-test-renderer';

import RosterItem from '../config/RosterItem.js';
import { rosters } from '../config/mockDataRoster.js';
import RosterPage from '../config/RosterPage.js';

it('renders a RosterPage using Snapshots', () => {
  const component = renderer.create(
    <RosterPage
      rosters={rosters}
      selectRoster={jest.fn}
    />
  );
  expect(component).toMatchSnapshot();
});
