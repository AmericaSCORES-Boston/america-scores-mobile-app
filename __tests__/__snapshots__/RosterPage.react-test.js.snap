exports[`test renders a RosterPage using Snapshots 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingVertical": 40,
    }
  }
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 8,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }>
          Test 1
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 8,
            }
          }>
          John Doe,Jane Doe stars
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 8,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }>
          Test 2
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 8,
            }
          }>
          Billy Bob stars
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 8,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "paddingBottom": 4,
            }
          }>
          Test 3
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 8,
            }
          }>
          John Michael,Harry Reynolds stars
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
